app/src/main/java/com/github/se/polyfit/viewmodel/post/CreatePostViewModel.kt

```diff
@@ -60,12 +76,14 @@ constructor(
     return _post.getProtein()?.amount ?: 0.0
   }
 
-  suspend fun setPost() {
-    try {
-      postFirebaseRepository.storePost(_post)
-    } catch (e: Exception) {
-      Log.e("CreatePostViewModel", "Failed to store post in the database : ${e.message}", e)
-      throw Exception("Failed to store post in the database : ${e.message}")
+  fun setPost() {
+    runBlocking {
```

Comment:
OK for now, but to avoid blocking the whole app in the future, we could launch a global coroutine to make the experience more fluid. Or we could just call that technical dept of the v1 that could be fixed in the v2.

Reply:
Yeah, that makes sense. It was to solve issue with threads too